axelrod/fingerprint.py

```diff
@@ -0,0 +1,206 @@
+import axelrod as axl
+import numpy as np
+import pandas as pd
+import seaborn as sns
+import matplotlib.pyplot as plt
+from itertools import product
+from axelrod.strategy_transformers import JossAnnTransformer, DualTransformer
+from axelrod.interaction_utils import compute_final_score_per_turn as cfspt
+from collections import OrderedDict
+
+
+class Fingerprint():
+    def __init__(self):
+        pass
+
+    def plot(self):
+        pass
+
+
+class AshlockFingerprint(Fingerprint):
+    def __init__(self, strategy, probe):
+        """
+        Parameters
+        ----------
+        strategy : class
+            A class that must be descended from axelrod.strategies.
+        probe : class
+            A class that must be descended from axelrod.strategies
+        """
+        self.strategy = strategy
+        self.probe = probe
+
+    def create_probe_coords(self, granularity):
+        """Creates a set of coordinates over the unit square.
+
+        Construncts (x, y) coordinates that are separated by a step equal to
+        `granularity`. The coordinates are over the unit squeare which implies
+        that the number of points created will be 1/`granularity`^2.
+
+        Parameters
+        ----------
+        granularity : float
+            The seperation between each coordinate. Smaller granularity will
+            produce more coordinates that will be closer together.
+
+        Returns
+        ----------
+        probe_coords : ordered dictionary
+            An Ordered Dictionary where the keys are tuples representing each
+            coordinate, eg. (x, y). The value is automatically set to `None`.
+        """
+        coordinates = list(product(np.arange(0, 1, granularity),
+                                   np.arange(0, 1, granularity)))
+
+        probe_coords = OrderedDict.fromkeys(coordinates)
```

Comment:
I don't understand why this needs to be an OrderedDict. Would a simple dictionary not do?